tutorials/language/tensorize.py

```diff
@@ -0,0 +1,276 @@
+"""
+Use Tensorize to Leverage Hardware Intrinsics
+=========
+**Author**: `Yizhi Liu <https://github.com/yzhliu>`_
+
+This is an introduction material on how to do tensorization in TVM.
+
+By using schedule primitive :code:`tensorize`,
+people can replace a unit of computation with the corresponding intrinsics,
+making it easy to leverage handcrafted micro-kernels,
+as well as extend TVM to support new hardware architectures.
+
+In this tutorial, we will demonstrate the usage of tensorization.
+Notice that the examples are more for showing the functionality,
+rather than providing an efficient solution.
+
+"""
+from __future__ import absolute_import, print_function
+
+import tvm
+import numpy as np
+
+######################################################################
+# Define Matrix Multiplication
+# ----------------------------
+# Take matrix multiplication as our example.
+# Matmul first multiply the corresponding elements between two matrix,
+# then accumulate across a certain axis.
+# The following lines describe the computation in TVM.
+#
+N, M, L = 1024, 512, 64
+A = tvm.placeholder((N, L), name='A')
+B = tvm.placeholder((M, L), name='B')
+k = tvm.reduce_axis((0, L), name='k')
+C = tvm.compute((N, M), lambda i, j:
+                tvm.sum(A[i, k] * B[j, k], axis=k), name='C')
+
+######################################################################
+# Schedule the Matmul
+# -------------------
+# Now suppose we had an accelerator supports matrix-vector multiplication (GEMV) as a hardware primitive,
+# which can take arbitrary size of reduce axis, but another axis needs to be no larger than 16.
+# Thus we break down the matmul loops to make the innermost loops a (16x64) GEMV.
+#
+s = tvm.create_schedule(C.op)
+factor = 16
+x, y = C.op.axis
+z, = C.op.reduce_axis
+yo, yi = s[C].split(y, factor=factor)
+s[C].reorder(x, yo, yi, z)
+print(tvm.lower(s, [A, B, C], simple_mode=True))
+
+######################################################################
+# As showed in the IR printed above,
+# the inner loops :code:`j.inner` along with :code:`k` together form a computation of GEMV
+# - within the inner most two loops, the index :code:`i` is fixed,
+# the access to the matrix :code:`A` only varies by :code:`k`,
+# which makes the access pattern of :code:`A` a "vector".
+# In order to leverage our "virtual" hardware's GEMV instruction,
+# we can tensorize over :code:`j.inner`.
+#
+# Define GEMV Tensorization Intrinsic
+# -----------------------------------
+# Before scheduling the tensorization, we need to first define the intrinsic function for GEMV.
+# It includes two parts, the first is computing definition of GEMV,
```

Comment:
I would reword to:
It includes two parts, the first is a compute definition of GEMV.